lib/libutee/arch/arm/user_ta_entry.c

```diff
@@ -104,17 +105,19 @@ static TEE_Result ta_header_add_session(uint32_t session_id)
 	if (ta_ref_count == 1) {
 		TEE_Result res;
 
-		if (!context_init) {
+		if (!init_done) {
 			trace_set_level(tahead_get_trace_level());
 			__utee_gprof_init();
 			malloc_add_pool(ta_heap, ta_heap_size);
 			_TEE_MathAPI_Init();
-			context_init = true;
 		}
-
-		res = TA_CreateEntryPoint();
-		if (res != TEE_SUCCESS)
-			return res;
+		if (!(ta_head.flags & TA_FLAG_INSTANCE_KEEP_ALIVE) ||
```

Comment:
Why do we test for the absence of `TA_FLAG_INSTANCE_KEEP_ALIVE` here? Isn't `!init_done` enough?

Reply:
Ah, right. `init_done` is per-instance already of course. Will fix.